pkg/storage/interfaces.go

```diff
@@ -167,4 +167,6 @@ type Interface interface {
 	//    }
 	// })
 	GuaranteedUpdate(ctx context.Context, key string, ptrToType runtime.Object, ignoreNotFound bool, precondtions *Preconditions, tryUpdate UpdateFunc) error
+
+	Xxx(ctx context.Context, key string, out runtime.Object, ignoreNotFound bool, precondtions *Preconditions, tryUpdate UpdateFunc, guess runtime.Object) error
```

Comment:
I also though about changing GuaranteedUpdate, but my thoughs were that it's better to add a new one instead of modifying existing because:
- we don't need to modify all calls to GuaranteedUpdate all over the codebase
- in the fullness of time, GuaranteedUpdate should be the only method in Cacher (the Xxx should only be in storage-specific implementation if at all).

However, I'm fine with changing GuaranteedUpdate if you prefer that.
@xiang90 - any opinion on that?